### ISSUE TITLE
[folderview]: mouse hover: icons sometimes hovered, sometimes not

### ISSUE THREAD

**Issue by star-buck** [blue-systems/plasma-5.9#85]
Plasma 5.4.9:
https://www.dropbox.com/s/2ae7tpljl7tn4mk/folderview-hover-sometimes.mkv?dl=0

Maybe it is due to the weird textline break?

**Comment by eikehein** [comment]
It's due to the weird text line break.

I did a selective backport of some improvements in 5.10 to the 5.9 branch that fix this:

https://commits.kde.org/plasma-desktop/59528fdc731afe5603bbca9f11c021b6d3415945

This will be in 5.9.5.

**Comment by star-buck** [comment]
Great, now does it actually fix the hover or the weird line break (or both)?

**Comment by eikehein** [comment]
From my testing it fixed both. As you suspected the hover problem was related to the line break.

I'll add though that the full 5.10 hover/selection etc. code does behave even nicer. We did a lot of tweaks for more natural feeling. Also moving icons by DND feels much more natural now because of smarter calculation of insert position.

Basically after we decided to make FV default in 5.10 (and even before that, my performance overhaul was started first) we put a ton of effort into making it nicer.
